### PR TITLE
CA-16/Fix floor plan crash

### PR DESCRIPTION
**Pull request by rock3r**
This PR addresses lemberg/connfa-android#16. It's another NPE, where a failing network request yields a null image source and then crashes the app. This PR includes the same kind of cleanup #19 contained, too. See commits for more details.

### CODE REVIEW

File: app/src/main/java/com/connfa/ui/fragment/FloorPlanFragment.java [hunk 1]
```diff
@@ -30,123 +30,176 @@
...
     private List<FloorPlan> plans;
     private SubsamplingScaleImageView floorImage;
+    private LoadPlansTask loadPlansTask;
+    private LoadPlanImageTask loadPlanImageTask;
 
-    private UpdatesManager.DataUpdatedListener updateListener = new UpdatesManager.DataUpdatedListener() {
-        @Override
-        public void onDataUpdated(List<Integer> requestIds) {
+    @Override
+    public void onStart() {
```
Comment (danybony) [inline app/src/main/java/com/connfa/ui/fragment/FloorPlanFragment.java]: Move this after `onCreateView` to follow the lifecycle

File: app/src/main/java/com/connfa/ui/fragment/FloorPlanFragment.java [hunk 2]
```diff
@@ -30,123 +30,176 @@
...
+        floorSelector.setAdapter(floorsAdapter);
 
-        return result;
+        floorSelector.setVisibility(floorPlans.isEmpty() ? View.INVISIBLE : View.VISIBLE);
     }
 
-    private class LoadPlansTask extends AsyncTask<Void, Void, List<FloorPlan>> {
+    private static class LoadPlansTask extends AsyncTask<Void, Void, List<FloorPlan>> {
+
+        private final LoadPlansTaskCallback callback;
```
Comment (danybony) [inline app/src/main/java/com/connfa/ui/fragment/FloorPlanFragment.java:149]: 👍

**Review by danybony — APPROVED** [review]
LGTM, just one method order comment

File: app/src/main/java/com/connfa/ui/fragment/FloorPlanFragment.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/com/connfa/ui/fragment/FloorPlanFragment.java]: 3f0c88e6e9a7d2014cd0becfeb935d2634bf664e

**Pull request merged**